### PR TITLE
Add ids to elements

### PR DESCRIPTION
**Pull request by yyanwang**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by yyanwang** [comment]
@jmfrancois @jsomsanith 
Hi, excuse me, but would you guys take a quick look on this? I hope I didn't misunderstand you in our previous discussion.
Now I only add ids to List component, If this approach is ok, I will continue on other components too.
Thanks!

File: src/List/DisplayTable/DisplayTable.component.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ function RowRenderer(props) {
 						<td key={index}>
 							{iconName && <Icon name={iconName} />}
 							<Button
+								id={`tc-list-row-${props.index}-col-${index}${props.id}`}
```
Comment (jsomsanith) [inline src/List/DisplayTable/DisplayTable.component.js]: The id is not meaningfull. By reading it, it seems to be targeting the cell, but in fact you want to target the row title.

File: src/List/Toolbar/Toolbar.component.js [hunk 2]
```diff
@@ -44,12 +46,15 @@ function Toolbar(props) {
 					/>) : null}
 					<ButtonGroup className="navbar-btn">
 						{props.listActions ? (
-							<DropdownButton id="tc-list-toolbar-add-dropdown" title="" bsStyle="link">
+							<DropdownButton
+								id={`tc-list-toolbar-add-dropdown${id}`}
```
Comment (jsomsanith) [inline src/List/Toolbar/Toolbar.component.js]: The id is seems to target something about `add` but it is a list of actions

**Comment by yyanwang** [comment]
@jsomsanith 
"... And this ensure the same meaning for the 3 display modes (yes, they should have the same ids)"
sorry but what do you mean by 3 display modes? what are "they"?

File: src/List/DisplayTable/DisplayTable.component.js [hunk 3]
```diff
@@ -73,14 +74,17 @@ ListHeader.propTypes = {
...
 	const className = classnames(
 		'table',
 		'tc-list-display-table',
 		theme.table,
 	);
+	id = id ? '-' + id : '';
 	return (
-		<table className={className}>
+		<table
+			id={`tc-list-table${id}`}
```
Comment (yyanwang) [inline src/List/DisplayTable/DisplayTable.component.js]: +1

File: src/List/Toolbar/Toolbar.component.js (hunk 2 above)
Comment (yyanwang) [inline src/List/Toolbar/Toolbar.component.js]: ok, will change this id.

File: src/List/DisplayTable/DisplayTable.component.js (hunk 1 above)
Comment (yyanwang) [inline src/List/DisplayTable/DisplayTable.component.js]: ok, will change this one.

**Comment by jsomsanith** [comment]
Yes I think the 3 displays should have the same ids on the same elements.

**Comment by yyanwang** [comment]
Changes have been moved to #99 .Please review/push to that one.

**Pull request closed without merging**